daemons/pacemakerd/pcmkd_subdaemons.c

```diff
@@ -194,24 +211,13 @@ check_active_before_startup_processes(gpointer user_data)
                                 pcmk_children[next_child].pid));
             }
             pcmk_process_exit(&(pcmk_children[next_child]));
-            if (!pcmk_children[next_child].respawn) {
-                /* if a subdaemon is down and we don't want it
-                   to be restarted this is a success during
-                   shutdown. if it isn't restarted anymore
-                   due to MAX_RESPAWN it is
-                   rather no success.
-                 */
-                if (pcmk_children[next_child].respawn_count <= MAX_RESPAWN) {
-                    subdaemon_check_progress = time(NULL);
-                }
-                next_child++;
-            }
```

Comment:
So with this change, we'll increment next_child when rc==pcmk_rc_ipc_unresponsive and the child needs to be respawned ... doesn't that mean we won't check it for ~10s?

Reply:
Yes. Guess that is good and bad. I was anyway torn between checking as soon as possible or giving it some time to get alive. And as the latter was now the easier one - regarding implementation - I decided to go with it.

Reply:
It makes me a little uncomfortable going that long without checking a newly respawned process, but I can't think of any reason why it would be a problem.

Reply:
It shouldn't make that much a difference as with the process disappearing we should get the signal immediately and if it is just ipc-unresponsive we anyway go 5 rounds before really reacting - so effectively we go just one more round like this.